### PR TITLE
Throw Exception if mandate is in pending state

### PR DESCRIPTION
**Pull request by ciungulete**

### CODE REVIEW

File: src/Exceptions/MandateIsNotYetFinalizedException.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class MandateIsNotYetFinalizedException extends Exception
+{
+    /**
+     * PlanNotFoundException constructor.
+     *
+     * @param string $message
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct(string $message = 'The customer mandate is in pending', int $code = 404, Throwable $previous = null)
```
Comment (sandervanhooft) [inline src/Exceptions/MandateIsNotYetFinalizedException.php:17]: "... is still pending"

File: tests/BaseTestCase.php [hunk 2]
```diff
@@ -310,6 +310,14 @@ protected function getMandatedUser($persist = true, $overrides = [])
         ], $overrides));
     }
 
+    protected function getPendingMandatedUser($persist = true, $overrides = [])
+    {
+        return $this->getCustomerUser($persist, array_merge([
+            'mollie_mandate_id' => 'mdt_unique_mandate_id',
+            'status' => 'pending',
```
Comment (sandervanhooft) [inline tests/BaseTestCase.php:317]: Shouldn't this be mollie_mandate_status?

**Review by sandervanhooft — CHANGES_REQUESTED** [review]
Nice first attempt.

File: tests/BaseTestCase.php [hunk 3]
```diff
@@ -310,6 +310,14 @@ protected function getMandatedUser($persist = true, $overrides = [])
         ], $overrides));
     }
 
+    protected function getPendingMandatedUser($persist = true, $overrides = [])
+    {
+        return $this->getCustomerUser($persist, array_merge([
+            'mollie_mandate_id' => 'mdt_unique_mandate_id',
+            'mollie_mandate_status' => 'pending',
```
[the remark this answers is not in the document]
Comment (ciungulete) [reply tests/BaseTestCase.php:317]: We need it in `throwExceptionIfMandateIsInPendintState`

**Review by sandervanhooft — APPROVED** [review]

**Pull request merged**